Persephone/AppDelegate.swift

```diff
@@ -6,84 +6,54 @@
 //  Copyright © 2018 Dan Barber. All rights reserved.
 //
 
-import Cocoa
+import AppKit
+import ReSwift
 import MediaKeyTap
 
 @NSApplicationMain
-class AppDelegate: NSObject, NSApplicationDelegate, MediaKeyTapDelegate {
-  var preferences = Preferences()
+class AppDelegate: NSObject,
+                   NSApplicationDelegate,
+                   MediaKeyTapDelegate {
   var mediaKeyTap: MediaKeyTap?
 
-  static let mpdClient = MPDClient(
-    withDelegate: NotificationsController()
-  )
-
   func applicationDidFinishLaunching(_ aNotification: Notification) {
-    connect()
-
-    preferences.addObserver(self, forKeyPath: "mpdHost")
-    preferences.addObserver(self, forKeyPath: "mpdPort")
+    App.mpdServerController.connect()
+    _ = App.userNotificationsController
```

Comment:
We need to touch this here because of the way Swift lazily loads static lets.

Reply:
Can you give me an example? I'm considering trying the coordinator pattern which I feel should clean this up. What do you think of that pattern?